test-result-summary-client/src/Build/GitNewIssue.jsx

```diff
@@ -73,31 +86,32 @@ export default class GitNewissue extends Component {
             });
         }
 
-        const buildStartTime = moment(parseInt(buildTimeStamp)).format(DAY_FORMAT);
+        const buildStartTime = moment(parseInt(timestamp)).format(DAY_FORMAT);
         const title = `${testName} ${testResult} in ${buildName}`;
         const nl = "\n";
-        const body = `**Test Info**${nl}`
+        const body = 
+            ((testName && duration && machine && testId) ? `**Test Info**${nl}`
             + `Test Name: ${testName}${nl}`
             + `Test Duration: ${renderDuration(duration)}${nl}`
             + `Machine: ${machine}${nl}`
             + `TRSS link for the test output: https://trss.adoptopenjdk.net/output/test${params({ id: testId })}${nl}`
-            + `${nl}${nl}`
-            + `**Build Info**${nl}`
+            + `${nl}${nl}` : ``)
+            + ((buildName && buildStartTime && buildUrl && buildId) ? `**Build Info**${nl}`
             + `Build Name: ${buildName}${nl}`
             + `Jenkins Build start time: ${buildStartTime}${nl}`
             + `Jenkins Build URL: ${buildUrl}${nl}`
             + `TRSS link for the build: https://trss.adoptopenjdk.net/allTestsInfo${params({ buildId: buildId })}${nl}`
-            + `${nl}${nl}`
-            + `**Java Version**${nl}`
-            + `${javaVersion}${nl}`
-            + ((!firstSeenFailure) ? `` : (
+            + `${nl}${nl}` : ``)
+            + (javaVersion ? `**Java Version**${nl}`
+            + `${javaVersion}${nl}` : ``)
+            + ((firstSeenFailure) ? (
                 `${nl}${nl}`
                 + `**This test has been failed ${failCount} times since ${moment(firstSeenFailure.timestamp).format(DAY_FORMAT)}**${nl}`
-                + `**Java Version when the issue first seen**${nl}`
-                + `${firstSeenFailure.javaVersion}${nl}`
+                + (firstSeenFailure.javaVersion ? `**Java Version when the issue first seen**${nl}`
+                + `${firstSeenFailure.javaVersion}${nl}` : ``)
                 + `Jenkins Build URL: ${firstSeenFailure.buildUrl}${nl}${nl}`
                 + failMachineUrlBody
-            ))
+            ) : `` )
             + (gitDiffLinksBody ? `${nl}**Git Diff of first seen failure and last success**${nl}` + gitDiffLinksBody : ``)
             + (rerunLink ? `${nl}[Rerun in Grinder](${rerunLink})` : ``);
 
```

Comment:
For maintenance purpose, can we separate this into mutiple variables?
const testInfo = (testName && duration && machine && testId) ? ...
const BuildInfo = ...
const javaVersionInfo = ...
const gitDiffLinksInfo =
const firstSeenFailureInfo = ...
const rerunLinkInfo = ...
const body = testInfo + BuildInfo ...